core/src/main/java/io/aiven/kafka/tieredstorage/manifest/serde/DataKeyDeserializer.java

```diff
@@ -19,22 +19,38 @@
 import javax.crypto.SecretKey;
 
 import java.io.IOException;
-import java.util.function.Function;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.function.BiFunction;
 
+import com.fasterxml.jackson.core.JsonParseException;
 import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.databind.DeserializationContext;
 import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
 
 public class DataKeyDeserializer extends StdDeserializer<SecretKey> {
-    private final Function<byte[], SecretKey> keyDecryptor;
+    private final BiFunction<String, byte[], SecretKey> keyDecryptor;
 
-    public DataKeyDeserializer(final Function<byte[], SecretKey> keyDecryptor) {
+    public DataKeyDeserializer(final BiFunction<String, byte[], SecretKey> keyDecryptor) {
         super(SecretKey.class);
-        this.keyDecryptor = keyDecryptor;
+        this.keyDecryptor = Objects.requireNonNull(keyDecryptor, "keyDecryptor cannot be null");
     }
 
     @Override
     public SecretKey deserialize(final JsonParser p, final DeserializationContext ctxt) throws IOException {
-        return keyDecryptor.apply(p.getBinaryValue());
+        final String keyText = p.getText();
+        final int colonI = keyText.indexOf(':');
+        if (colonI < 0) {
+            throw new JsonParseException("Malformed encrypted data key string: " + keyText);
+        }
+        // Sanity check for more than one ':'
+        final int lastColonI = keyText.lastIndexOf(':');
+        if (lastColonI != colonI) {
+            throw new JsonParseException("Malformed encrypted data key string: " + keyText);
+        }
+
+        final String keyEncryptionKeyId = keyText.substring(0, colonI);
+        final String dataKeyBase64 = keyText.substring(colonI + 1);
```

Comment:
What about moving this into `DataEncryptionKeyResult` instead?
Then we could have something like: 
```suggestion
        final var result = DataEncryptionKeyResult.decode(p.getText());
```

And BiFunction could be back to `Function<DataEncryptionKeyResult, byte[]>`

Reply:
Yes, done this. Also renamed it to `EncryptedDataKey`